### ISSUE TITLE
Move the connectors in a service worker

### ISSUE THREAD

**Issue by TitanNano** [loqui/im#813]
This will be a huge step, but as far as I know we will get ServiceWorker with FxOS 2.5. We should detect if we can uses them and if available do the connecting part in the worker, so eventually we are able to quit the main application and still being connected to our services.

**Comment by gargsms** [comment]
Ah. That is very nice.
Also, we should leverage normal WebWorker for some of the computationally expensive operations like reading from the Store, or computing the authentications hash so that we don't block the main thread and lead to a bad user experience.

I need to read about Service Workers. Currently total n00b.

**Comment by TitanNano** [comment]
yes, what I'd like to see is, that we take care of everything, from network connection over reading/writeing the storage to handling chats and accounts, inside of a worker and leave just the UI handling to the main thread.
